### PR TITLE
Ensure that _remotes_ is always the most recent version

### PR DESCRIPTION
**Pull request by pat-s**
I just faced a very ugly situation on one of the older rocker/verse images which come with an outdated version of the _remotes_ pkg.

Using only `if (!requireNamespace("remotes")` won't trigger an install all subsequent steps will use an outdated remotes version.

This finally triggered a build error when calling `remotes::install_deps()` with remotes v1.1.0 in a R 3.4.4 container.

This fix ensures that _remotes_ is always updated if it is already installed.

Strictly we would not need this change on Travis but for consistency...

### CODE REVIEW

File: .circleci/config.yml [hunk 1]
```diff
@@ -18,7 +18,7 @@ jobs:
           name: Install package dependencies
           command: |
             echo -e 'options(Ncpus = 6, repos = structure(c(CRAN = "https://cloud.r-project.org/")))' > $HOME/.Rprofile
-            R -q -e 'if (!requireNamespace("remotes")) install.packages("remotes")'
+            R -q -e 'if (!requireNamespace("remotes") || "remotes" %in% old.packages()) install.packages("remotes")'
```
Comment (krlmlr) [inline .circleci/config.yml]: `old.packages()` is slow, we only need to check {remotes}.

```suggestion
            R -q -e 'if (!requireNamespace("remotes") || packageVersion("remotes") < "x.y.z") ...'
```

Comment (pat-s) [reply .circleci/config.yml]: We need to make this dynamic on the latest release version of _remotes_.
Is there a better way than the following overkill?
I don't want to have this in the YAML.

``` r
desc = readLines(url("https://raw.githubusercontent.com/r-lib/remotes/master/DESCRIPTION")) 
desc2 = desc[grep("Version:", desc)]
desc3 = strsplit(desc2, " ")[[1]][2]
substr(desc3, 1, 5)
#> [1] "2.1.0"
```

<sup>Created on 2019-12-01 by the [reprex package](https://reprex.tidyverse.org) (v0.3.0)</sup>

Comment (krlmlr) [reply .circleci/config.yml]: What about `install.packagges("remotes")` ? Sound like the least of all evils.

Comment (pat-s) [reply .circleci/config.yml]: Costs us some seconds but yeah, maybe this is the simplest way to ensure max safety in this issue.

**Pull request merged**